platform-coop-toolkit/coop-library-framework
# feat: migrate fields to ACF Pro

## Description

Migrates custom fields to ACF Pro.

- [x] Create new fields.
- [x] Customize the `day` field based on selected `year` and `month`.
- [x] Save publication date if applicable.
- [x] Add validation for Perma.cc and Internet Archive links.
- [x] Add validation for DOI.
- [x] Add validation for ISBN.
- [x] Add validation for ISSN.
- [x] Convert existing authors, editors, and translators.
- [x] Convert existing Perma.cc and Internet Archive links.

## Steps to test

1. Check out this branch.
2. Update Composer dependencies: `composer install`.
3. Update NPM dependencies and build assets: `npm ci && npm run build`.
4. Install and activate Advanced Custom Fields Pro.
5. Run the [metadata conversion script](https://github.com/platform-coop-toolkit/coop-library-framework/blob/a02117529f10e87976152bf2eebbcb48323a3bab/bin/README.md).
5. Visit resource editor, test editing UI and user experience while modifying all fields.

**Expected behavior:** Validation is handled according to expectations and appropriate messages are displayed to users. Authors appear as expected in new author fields.

## Additional information

Not applicable.

## Related issues

- Resolves #307.
